scala/lambdas/custodial-copy-queue-creator/src/main/scala/uk/gov/nationalarchives/custodialcopyqueuecreator/Lambda.scala

```diff
@@ -51,30 +50,36 @@ object Lambda:
   case class Config(secretName: String, outputQueue: String) derives ConfigReader
   case class Dependencies(entityClient: EntityClient[IO, Fs2Streams[IO]], sqsClient: DASQSClient[IO], uuidGenerator: () => UUID)
 
-  private def toJson(id: UUID, deleted: Boolean, messageType: String): Json =
-    Json.fromJsonObject(JsonObject(("id", Json.fromString(s"$messageType:$id")), ("deleted", Json.fromBoolean(deleted))))
+  private def toJson(id: UUID, deleted: Boolean, potentialMessageType: Option[String]): Json = {
+    val idValue = potentialMessageType.map(messageType => s"$messageType:$id").getOrElse(id.toString)
+    Json.fromJsonObject(JsonObject(("id", Json.fromString(idValue)), ("deleted", Json.fromBoolean(deleted))))
+  }
 
   given Encoder[MessageBody] =
-    case IoMessageBody(id, deleted) => toJson(id, deleted, "io")
-    case CoMessageBody(id, deleted) => toJson(id, deleted, "co")
-    case SoMessageBody(id, deleted) => toJson(id, deleted, "so")
+    case IoMessageBody(id)      => toJson(id, false, "io".some)
+    case CoMessageBody(id)      => toJson(id, false, "co".some)
+    case SoMessageBody(id)      => toJson(id, false, "so".some)
+    case DeletedMessageBody(id) => toJson(id, true, None)
 
   given Decoder[MessageBody] = (c: HCursor) =>
     for {
       id <- c.downField("id").as[String]
       deleted <- c.downField("deleted").as[Boolean]
     } yield {
       val typeAndRef = id.split(":")
-      val ref = UUID.fromString(typeAndRef.last)
-      val entityType = typeAndRef.head
-      entityType match {
-        case "io" => IoMessageBody(ref, deleted)
-        case "co" => CoMessageBody(ref, deleted)
-        case "so" => SoMessageBody(ref, deleted)
-      }
+      typeAndRef match
+        case Array(entityType, refString) =>
+          val ref = UUID.fromString(refString)
+          entityType match {
+            case "io" => IoMessageBody(ref)
+            case "co" => CoMessageBody(ref)
+            case "so" => SoMessageBody(ref)
+          }
+        case Array(ref) => DeletedMessageBody(UUID.fromString(ref))
     }
 
   enum MessageBody(val id: UUID, val deleted: Boolean):
-    case IoMessageBody(override val id: UUID, override val deleted: Boolean) extends MessageBody(id, deleted)
-    case CoMessageBody(override val id: UUID, override val deleted: Boolean) extends MessageBody(id, deleted)
-    case SoMessageBody(override val id: UUID, override val deleted: Boolean) extends MessageBody(id, deleted)
+    case IoMessageBody(override val id: UUID) extends MessageBody(id, false)
+    case CoMessageBody(override val id: UUID) extends MessageBody(id, false)
+    case SoMessageBody(override val id: UUID) extends MessageBody(id, false)
+    case DeletedMessageBody(override val id: UUID) extends MessageBody(id, true)
```

Comment:
Maybe `DeletedEntityMessageBody`?

Reply:
It's a bit long. I'll change it to `DeletionMessageBody` to match the change on the CC PR.